mini-frame.el

```diff
@@ -283,10 +287,11 @@ ALIST is passed to `window--display-buffer'."
                     (eq ignored-command this-command))
               (throw 'ignored t)))))
     (apply fn args))
-   ((and (frame-live-p mini-frame-frame)
-         (frame-visible-p mini-frame-frame))
-    (mini-frame--display fn args))
```

Comment:
Fixed

Reply:
But why this block must be removed at all? Is this Windows specific? Can you please provide receipt for issue that this removal is fixing?

Reply:
What this block doing is just call bare `mini-frame--display` without any additional setup + cleanup.
That can be quite dangerous.

For example when I tried it on Windows, once the minibuffer some how got into visible state, and the focus is removed. Since we never run the cleanup code for it, it will just stays visible there forever.
This could be true even for other platform too.

Reply:
> What this block doing is just call bare `mini-frame--display` without any additional setup + cleanup.
> That can be quite dangerous.

But mini-frame is visible. This mean that cleanup is set already and all what we want from`mini-frame--display` is maybe to update size and position of mini-frame. 

> For example when I tried it on Windows, once the minibuffer some how got into visible state, and the focus is removed. Since we never run the cleanup code for it, it will just stays visible there forever.
> This could be true even for other platform too.

I'll check it on Windows.

Reply:
> But mini-frame is visible. This mean that cleanup is set already and all what we want from`mini-frame--display` is maybe to update size and position of mini-frame.

Let me clarify it more.
I found out one of the reason in Windows (or it may affect other GUI system) that some time the mini frame may get behind the main frame.
In that case the mini frame still show up as visible but user cannot see it.
Worse, I observe that when I'm using `swiper`, once the mini frame got behind main frame, the user will not be able to move away from mini frame without using mouse.
The reason may be because the mini frame is still visible (actually hidden since it's behind the main frame), so when we try to quit it, there is no cleanup code to be run to reassign the input focus.

To recover from the issue of mini frame get behind main frame, one easy solution is to just hide mini frame every time we enter it (except recursive entering). That will make sure we start it with a fresh z-order once it went wrong. However, because of that, we will need the frame cleanup code for the your mentioned block, since basically we're going to show the mini frame again.

Reply:
> Worse, I observe that when I'm using `swiper`, once the mini frame got behind main frame, the user will not be able to move away from mini frame without using mouse.

`C-x 5 0` may help.

Can confirm. When `ivy-mode` is active or after `M-x swiper` mini-frame has `visibility` frame parameter set to `t` but it's not visible, e.g.:

1. `M-x swiper`
2. `C-g`
3. `M-: (frame-parameter mini-frame-frame 'visibility)`

I'll look further.